sympy/core/power.py

```diff
@@ -71,6 +71,9 @@ def integer_nthroot(y, n):
         raise ValueError("y must be nonnegative")
     if n < 1:
         raise ValueError("n must be positive")
+    if HAS_GMPY:
+        x, t = gmpy.iroot(y, n)
+        return as_int(x), t
```

Comment:
I would suggest splitting out everything below this into a private helper function so that it can accessed directly if so desired.

Reply:
Yes, I will change it.